kani-driver/src/call_goto_instrument.rs

```diff
@@ -116,19 +131,45 @@ impl KaniSession {
     }
 
     /// Generate a .c file from a goto binary (i.e. --gen-c)
-    pub fn gen_c(&self, file: &Path) -> Result<()> {
-        let output_filename = alter_extension(file, "c");
-        // We don't put the C file into temporaries to be deleted.
-
+    pub fn gen_c(&self, file: &Path, output_file: &Path) -> Result<()> {
         let args: Vec<OsString> = vec![
             "--dump-c".into(),
             file.to_owned().into_os_string(),
-            output_filename.into_os_string(),
+            output_file.to_owned().into_os_string(),
         ];
 
         self.call_goto_instrument(args)
     }
 
+    /// Generate a .demangled.c file from the .c file using the `prettyName`s from the symbol tables
+    pub fn demangle_c(
+        &self,
+        symtab_files: &[impl AsRef<Path>],
+        c_file: &Path,
+        demangled_file: &Path,
+    ) -> Result<()> {
+        let c_code = std::fs::read_to_string(c_file)?;
+        let mut demangled_code = c_code.clone();
+        for symtab_file in symtab_files {
+            let reader = BufReader::new(File::open(symtab_file.as_ref())?);
+            let symtab: serde_json::Value = serde_json::from_reader(reader)?;
+            if let Some(symtab) = symtab.get("symbolTable").and_then(|s| s.as_object()) {
+                for (_, symbol) in symtab {
+                    if let Some(serde_json::Value::String(name)) = symbol.get("name") {
+                        if let Some(serde_json::Value::String(pretty)) = symbol.get("prettyName") {
+                            let name = name.strip_prefix("tag-").unwrap_or(name);
```

Comment:
Struct names start with `tag-` but in the GotoC file, the "tag-" prefix is not there, so we want to get rid of it. On the other hand, not all names start with "tag-", so we only strip it if necessary, and otherwise leave it (-> unwrap_or).

Reply:
Comment explaining that